acquia/mc-cs-plugin-custom-objects
# MAUT-7886 Import custom item using unique hash

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ No]
| New feature/enhancement? (use the a.x branch)      | [ Yes]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #[MAUT-7886](https://backlog.acquia.com/browse/MAUT-7886)... 



#### Description:



#### Steps to test this PR:


1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a Custom Object "Book/Books"
3. Add a text field "Author" and mark it as a unique identifier field
4. Now create a custom Item for Books. Name: Factfulness, Author: Hans Rosling
5. Create another custom item with Name: How I learned to understand the world, Author: Hans Rosling
6. The previous custom item should have been replaced.  
7. Create another book now. This time, Name: Origin, Author: Dan Brown.
8. A new book should be created .
9. Try linking contacts with the custom items and try linking custom items with custom items. It should work fine.
10. Import custom items for books. it should not allow creation of duplicate authors. And book with a duplicate author should be updated with new values
